### DISCUSSION TITLE
Gui for Behave

### DISCUSSION THREAD

**Discussion by fallwest** [behave/behave#934] [Ideas]
Hi!

I have been enjoying Behave for couple years now, testing reverse vending machines that take back your empty bottles. I've written a Gui that helps the less technical members of my team find existing test step implementations and incorporate them in their tests. This solution also shows existing usage examples, to help them understand how to use the test step. Moreover, the test step implementation Python functions are displayed. There's also the possibility to run the test right from the gui with a full console log.  All of this is made possible by the rich metadata we get from Behave (which we think is great). We think the solution works better than the only commercially available product in the same space. We are thinking about open sourcing the code. If there is any interest, please get in touch: james.vestfall@gmail.com
Screenshot: 
![btc](https://user-images.githubusercontent.com/16136034/123930202-ceeb2600-d98f-11eb-98d9-19b672eef733.png)

**Comment by bittner** [comment]
Sure! That would be a great contributor project. Do you want me to create a repository for you in https://github.com/behave-contrib/?

**Reply by fallwest** [reply]
Hi! I'm still discussing this with my employer. But that sounds like a good plan. I think it would be good for all involved to be under your contrib folder. I'll come back to you as soon as I get a green light.

**Comment by fallwest** [comment]
Hi Peter!
I have now secured the required approvals from my employer to open-source the project. If you can create a repository under  https://github.com/behave-contrib/, I can check the code in there. Our working name for the project is "btcgui", where btc stands for "Behave Test Constellation". But in line with the other Contrib projects, we could also call it behave-gui. Let me know what you think is best.
James Westfall

**Reply by bittner** [reply]
There you go! https://github.com/behave-contrib/behave-gui

I sent you an invite into the group of maintainers. You should be able to push code to the project. If not let us know!

**Reply by bittner** [reply]
Looking at the screenshot above I feel that "GUI" is maybe misleading. At least for me "Graphical User Interface" has a tone of something that runs "natively" on a graphical desktop environment, in the sense of "not in a Web browser". Not sure if "frontend" better conveys the Web technology aspect or not.

**Reply by fallwest** [reply]
I've pushed to the project now. Take a look. I'm fine with changing the name to behave-frontend, but don't think I have rights to do so. I also welcome any other feedback!
